libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -569,6 +569,13 @@ void Compass::_detect_backends(void)
         } \
     } while (0)
 
+/*for chibios board coniguration */
```

Comment:
this would be better down further in the file, with the other HAL_COMPASS_DEFAULT == HAL_COMPASS_xxx checks I think